src/pages/billing/NewBillingProjectWizard/AzureBillingProjectWizard/AzureBillingProjectWizard.ts

```diff
@@ -0,0 +1,160 @@
+import _ from 'lodash/fp'
+import { Fragment, ReactNode, useEffect, useState } from 'react'
+import { h } from 'react-hyperscript-helpers'
+import { spinnerOverlay } from 'src/components/common'
+import { Ajax } from 'src/libs/ajax'
+import { reportErrorAndRethrow } from 'src/libs/error'
+import Events from 'src/libs/events'
+import { summarizeErrors, withBusyState } from 'src/libs/utils'
+import * as Utils from 'src/libs/utils'
+import { billingProjectNameValidator } from 'src/pages/billing/List'
+import { AzureManagedAppCoordinates } from 'src/pages/billing/models/AzureManagedAppCoordinates'
+import { BillingRole } from 'src/pages/billing/models/BillingProject'
+import { AddUsersStep } from 'src/pages/billing/NewBillingProjectWizard/AzureBillingProjectWizard/AddUsersStep'
+import { AzureSubscriptionStep } from 'src/pages/billing/NewBillingProjectWizard/AzureBillingProjectWizard/AzureSubscriptionStep'
+import { CreateNamedProjectStep } from 'src/pages/billing/NewBillingProjectWizard/AzureBillingProjectWizard/CreateNamedProjectStep'
+import { StepWizard } from 'src/pages/billing/NewBillingProjectWizard/StepWizard/StepWizard'
+import { validate } from 'validate.js'
+
+
+interface AzureBillingProjectWizardProps {
+  onSuccess: (string) => void
+}
+
+export const userInfoListToProjectAccessObjects = (emails: string, role: BillingRole): Array<{email: string; role: BillingRole}> => {
+  if (emails.trim() === '') {
+    return []
+  } else {
+    return _.flatten(emails.split(',').map(email => ({ email: email.trim(), role })))
+  }
+}
+
+export const AzureBillingProjectWizard = ({ onSuccess }: AzureBillingProjectWizardProps) => {
+  const [activeStep, setActiveStep] = useState<number>(1)
+  const [subscriptionId, setSubscriptionId] = useState<string>('')
+  // undefined used to indicate that the user has not yet typed in the input (don't want to show error)
+  const [billingProjectName, setBillingProjectName] = useState<string|undefined>(undefined)
+  const [userEmails, setUserEmails] = useState({ emails: '', hasError: false })
+  const [ownerEmails, setOwnerEmails] = useState({ emails: '', hasError: false })
+  // undefined used to indicate that the user has not yet made a selection
+  const [addUsersOrOwners, setAddUsersOrOwners] = useState<boolean|undefined>(undefined)
+  const [managedApp, setManagedApp] = useState<AzureManagedAppCoordinates>()
+
+  const [existingProjectNames, setExistingProjectNames] = useState<string[]>([])
+  const [projectNameErrors, setProjectNameErrors] = useState<ReactNode>()
+
+  const [isBusy, setIsBusy] = useState(false)
+
+  const createBillingProject = _.flow(
```

Comment:
A brought much of this code over from `CreateAzureBillingProjectModal`. I gave up on using `useLoadedState` because I wasn't sure how to handle error conditions (and I'm just not familiar with it yet).

Reply:
Better done than perfect.
But you can just render errors when the state of the loaded data is `'Error'`:
```typescript
const [data, updateData] = useLoadedData(updateDataMethod)

return .... 
  // render error
  data.status === 'Error' &&  // error component 

```

Reply:
I will eventually get used to `useLoadedData`, but at this point I'm not a huge fan. One of the things that I find confusing is that it's less obvious if all the ajax calls have error handling around them.